wireless_modules/mqtt_test/fake.py

```diff
@@ -0,0 +1,156 @@
+import ujson
+import urandom
+
+class module:
+    def __init__(self, module_num):
+        """ module_num what sensors to use"""
+        self.sensor = Sensor()
+
+        # list of all of the fake sensors in each fasdfa
+        if module_num == "1":
+            self.sensor_lst = ["temperature", "humidity", "steering_angle"]
+        
+        elif module_num == "2":
+            self.sensor_lst = ["co2", "temperature", "humidity", "accelerometer", "gyroscope"]
+        
+        elif module_num == "3":
+            self.sensor_lst = ["co2", "reed_velocity", "gps"]
+        
+        elif module_num == "4":
+            self.sensor_lst = ["power", "cadence", "heart_rate"]
+
+                
+        # Fake battery starts at 100%
+        self.battery_pc = 100 
+
+    def data(self):
+        data_dict = {"sensors": []}
+
+        for data in self.sensor_lst:
+            if data == "temperature":
+                data_dict["sensors"].append(self.sensor.temperature())
+
+            elif data == "humidity":
+                data_dict["sensors"].append(self.sensor.humidity())
+
+            elif data == "steering_angle":
+                data_dict["sensors"].append(self.sensor.steering_angle())
+
+            elif data == "co2":
+                data_dict["sensors"].append(self.sensor.co2())
+
+            elif data == "accelerometer":
+                data_dict["sensors"].append(self.sensor.accelerometer())
+                
+            elif data == "gyroscope":
+                data_dict["sensors"].append(self.sensor.gyroscope())
+                
+            elif data == "reed_velocity":
+                data_dict["sensors"].append(self.sensor.reed_velocity())
+                
+            elif data == "gps":
+                data_dict["sensors"].append(self.sensor.gps())
+                
+            elif data == "power":
+                data_dict["sensors"].append(self.sensor.power())
+                
+            elif data == "cadence":
+                data_dict["sensors"].append(self.sensor.cadence())
+                
+            elif data == "heart_rate":
+                data_dict["sensors"].append(self.sensor.heart_rate())
+
+        return ujson.dumps(data_dict)
+
+    def battery(self):
+        """ Decrease battery % by 1 and convert to JSON string """
+        self.battery_pc -= 1
+        battery_dict = {"percentage": self.battery_pc}
+        return ujson.dumps(battery_dict)
+
+
+
+class Sensor:
```

Comment:
If we wanted this class to be proper, we should have a `Sensor` interface in which a `TemperatureSensor` will implement. Currently your class describes a sensor with many types of sensors which doesn't make sense. This is probably unnecessary because it's fake so a better implementation could be a `FakeSensor` class where you include the type as a parameter into the constructor. Example: `FakeSensor("temperature")`. 

An even more elegant solution would be to have a generic `FakeSensor` class where you input a string AND `Value` class. Example:`FakeSensor("temperature", RandomValue)` or `FakeSensor("accelerometer", XYZValue)`.

Feel free to choose! There's just a lot of duplicate code and don't think we should ignore it even if it is fake. This is probably what we will end up using too!

Reply:
Yeah 100%. I kinda hacked it together so I will try and implement it correctly